proseco/acq.py

```diff
@@ -848,6 +936,45 @@ def calc_p_brightest(self, acq, box_size, man_err=0, bgd=0):
 
         return prob
 
+    def get_overlap_penalties(self):
+        """
+        Get the probability penalties for overlapping boxes.
+
+        This is returned as a multiplicative factor on the acquisition success
+        probability. No overlap penalty implies a value of 1.0.
+
+        :returns: list of penalties (float)
+        """
+        n_acq = len(self)
+        penalties = np.ones(n_acq)
+        if os.environ.get("PROSECO_DISABLE_OVERLAP_PENALTY") == "True":
+            return penalties
+
+        mags = self["mag"].data
+        halfws = self["halfw"].data
+        yangs = self["yang"].data
+        zangs = self["zang"].data
+
+        for idx1 in range(n_acq):
+            mag1 = mags[idx1]
+            halfw1 = halfws[idx1]
+            yang1 = yangs[idx1]
+            zang1 = zangs[idx1]
+
+            for idx2 in range(idx1 + 1, n_acq):
+                mag2 = mags[idx2]
+                halfw2 = halfws[idx2]
+                yang2 = yangs[idx2]
+                zang2 = zangs[idx2]
+                if box_overlap(yang1, zang1, halfw1, yang2, zang2, halfw2):
+                    if mag1 + OVERLAP_MAG_DEADBAND < mag2:
+                        # Star 1 is at least 0.2 mag brighter than star 2
+                        penalties[idx1] = OVERLAP_P_ACQ_PENALTY
```

Comment:
I note that from inspection and a quick functional test, it looks like there's no code here to *increase* the penalty on more than one overlap (because the penalty is just set with `=` not `*=`, right?).  I think that's completely fine but not sure if it is worth a comment.